extensions/binding/org.eclipse.smarthome.binding.lifx/src/main/java/org/eclipse/smarthome/binding/lifx/internal/LifxLightCurrentStateUpdater.java

```diff
@@ -123,39 +124,47 @@ private void sendLightStateRequests() {
         GetRequest statePacket = new GetRequest();
         communicationHandler.sendPacket(statePacket);
 
-        if (thingTypeUID.equals(THING_TYPE_COLORIRLIGHT)) {
+        if (product.isInfrared()) {
             GetLightInfraredRequest infraredPacket = new GetLightInfraredRequest();
             communicationHandler.sendPacket(infraredPacket);
         }
+        if (product.isMultiZone()) {
+            GetColorZonesRequest colorZonesPacket = new GetColorZonesRequest();
+            communicationHandler.sendPacket(colorZonesPacket);
+        }
     }
 
     @Override
     public void handleResponsePacket(Packet packet) {
-        if (packet instanceof StateResponse) {
-            handleLightStatus((StateResponse) packet);
-        } else if (packet instanceof StatePowerResponse) {
-            handlePowerStatus((StatePowerResponse) packet);
-        } else if (packet instanceof StateLightPowerResponse) {
-            handleLightPowerStatus((StateLightPowerResponse) packet);
-        } else if (packet instanceof StateLightInfraredResponse) {
-            handleInfraredStatus((StateLightInfraredResponse) packet);
-        }
+        try {
+            lock.lock();
 
-        if (currentLightState.isOnline() && !wasOnline) {
-            wasOnline = true;
-            logger.trace("{} : The light just went online, immediately polling the state of the light", macAsHex);
-            sendLightStateRequests();
+            if (packet instanceof StateResponse) {
+                handleLightStatus((StateResponse) packet);
+            } else if (packet instanceof StatePowerResponse) {
+                handlePowerStatus((StatePowerResponse) packet);
+            } else if (packet instanceof StateLightPowerResponse) {
+                handleLightPowerStatus((StateLightPowerResponse) packet);
+            } else if (packet instanceof StateLightInfraredResponse) {
+                handleInfraredStatus((StateLightInfraredResponse) packet);
+            } else if (packet instanceof StateMultiZoneResponse) {
+                handleMultiZoneStatus((StateMultiZoneResponse) packet);
+            }
+
+            if (currentLightState.isOnline() && !wasOnline) {
+                wasOnline = true;
+                logger.trace("{} : The light just went online, immediately polling the state of the light", macAsHex);
+                sendLightStateRequests();
+            }
+        } catch (Exception e) {
```

Comment:
I've removed the catch because the lock is anyhow released in the finally clause. There were no checked exceptions to catch.